### PR TITLE
feat: Upgrade nodeContains utility

### PR DESCRIPTION
**Pull request by just-boris**
*Issue #, if available:*

Porting changes from https://github.com/cloudscape-design/components/pull/1374

*Description of changes:*

* Added support for `EventTarget` arguments to allow `nodeContains(element, event.target)` without type issues
* Removed fallback, because in our supported browsers the native code works
* Updated tests

Note, the utility is still not used inside our components. I will open a follow up PR with migration

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/dom/__tests__/node-contains.test.ts [hunk 1]
```diff
@@ -10,61 +10,52 @@ test('returns false if either node is null', () => {
...
-              <div id="inbetween">
-                <div id="child"></div>
-              </div>
-            </div>
+    <div id="inbetween">
+      <div id="child"></div>
+    </div>
+  </div>
 `,
-  svg: `<g id="parent">
```
Comment (just-boris) [inline src/dom/__tests__/node-contains.test.ts:20]: This test did not check what it says. `<g>` on its own is not an SVG element, only inside `<svg>`, it becomes special

File: src/dom/__tests__/node-contains.test.ts [hunk 2]
```diff
@@ -10,61 +10,52 @@ test('returns false if either node is null', () => {
...
+    const div = document.createElement('div');
+    /* eslint-disable-next-line no-unsanitized/property */
+    div.innerHTML = testCases[title];
+
+    const parent = div.querySelector('#parent');
+    const inbetween = div.querySelector('#inbetween');
+    const child = div.querySelector('#child');
+
+    expect(nodeContains(parent, parent)).toBe(true);
+    expect(nodeContains(child, child)).toBe(true);
```
Comment (just-boris) [inline src/dom/__tests__/node-contains.test.ts:54]: Added missing coverage for check that the node contains itself

File: src/dom/__tests__/node-contains.test.ts [hunk 3]
```diff
@@ -10,61 +10,52 @@ test('returns false if either node is null', () => {
...
+    expect(nodeContains(parent, child)).toBe(true);
+    expect(nodeContains(parent, inbetween)).toBe(true);
+    expect(nodeContains(inbetween, child)).toBe(true);
+    expect(nodeContains(child, parent)).toBe(false);
+    expect(nodeContains(child, inbetween)).toBe(false);
```
Comment (jperals) [inline src/dom/__tests__/node-contains.test.ts:59]: For completeness we could also add:

```
expect(nodeContains(inbetween, parent)).toBe(false);
```

**Review by jperals — APPROVED** [review]

File: src/dom/__tests__/node-contains.test.ts (hunk 3 above)
Comment (just-boris) [reply src/dom/__tests__/node-contains.test.ts:59]: done

**Review by jperals — APPROVED** [review]

**Pull request merged**